test/export/StructureDefinitionExporter.test.ts

```diff
@@ -2294,6 +2410,620 @@ describe('StructureDefinitionExporter', () => {
+      const rootFlag = new FlagRule('component.interpretation');
+      rootFlag.mustSupport = true;
+      const labFlag = new FlagRule('component[Lab].interpretation');
+      labFlag.mustSupport = false;
+      observationWithSlice.rules.push(rootFlag, labFlag);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootInterpretation = sd.findElement('Observation.component.interpretation');
+      const labInterpretation = sd.findElement('Observation.component:Lab.interpretation');
+      expect(rootInterpretation.mustSupport).toBe(true);
+      expect(labInterpretation.mustSupport).toBe(true);
+    });
+
+    // don't actually apply the rule to the slice, since it might be valid based on the contents of the value sets
+    // but if it's the same value set for both, _then_ we know it can't be a weaker binding, and that's what we should give a message about
+    // so this is ok:
+    // * category[Procedure] from ABC (extensible)
+    // * category from ABC (required)
+    // and it will update the strength to required
+    // this is not:
+    // * category from ABC (required)
+    // * category[Procedure] from ABC (extensible)
+
+    it('should apply ValueSetRules on a slice, then a sliced element, with different value sets', () => {
+      // * category[Procedure] from http://example.com/MediocreObservationCodes (extensible)
+      // * category from http://example.com/ImportantObservationCodes (required)
+      const procedureValueSet = new ValueSetRule('category[Procedure]');
+      procedureValueSet.valueSet = 'http://example.com/MediocreObservationCodes';
+      procedureValueSet.strength = 'extensible';
+      const rootValueSet = new ValueSetRule('category');
+      rootValueSet.valueSet = 'http://example.com/ImportantObservationCodes';
+      rootValueSet.strength = 'required';
+
+      observationWithSlice.rules.push(procedureValueSet, rootValueSet);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootCategory = sd.findElement('Observation.category');
+      const procedureCategory = sd.findElement('Observation.category:Procedure');
+      const importantBinding = {
+        valueSet: 'http://example.com/ImportantObservationCodes',
+        strength: 'required'
+      };
+      const mediocreBinding = {
+        valueSet: 'http://example.com/MediocreObservationCodes',
+        strength: 'extensible'
+      };
+      expect(rootCategory.binding).toEqual(importantBinding);
+      expect(procedureCategory.binding).toEqual(mediocreBinding);
+    });
+
+    it('should apply ValueSetRules on a sliced element, then a slice, with different value sets', () => {
+      // * category[Procedure] from http://example.com/MediocreObservationCodes (extensible)
+      // * category from http://example.com/ImportantObservationCodes (required)
```

Comment:
This comment looks like it doesn't match what the rules are doing. This is the wrong order right?

Reply:
Correct! I'll fix the comment to reflect the actual test being performed.